test/lint/lint-python.py

```diff
@@ -0,0 +1,134 @@
+    'F405,'  # foo_function may be undefined, or defined from star imports: bar_module
+    'F406,'  # "from module import *" only allowed at module level
+    'F407,'  # an undefined __future__ feature name was imported
+    'F601,'  # dictionary key name repeated with different values
+    'F602,'  # dictionary key variable name repeated with different values
+    'F621,'  # too many expressions in an assignment with star-unpacking
+    'F622,'  # two or more starred expressions in an assignment (a, *b, *c = d)
+    'F631,'  # assertion test is a tuple, which are always True
+    'F632,'  # use ==/!= to compare str, bytes, and int literals
+    'F701,'  # a break statement outside of a while or for loop
+    'F702,'  # a continue statement outside of a while or for loop
+    'F703,'  # a continue statement in a finally block in a loop
+    'F704,'  # a yield or yield from statement outside of a function
+    'F705,'  # a return statement with arguments inside a generator
+    'F706,'  # a return statement outside of a function/method
+    'F707,'  # an except: block as not the last exception handler
+    'F811,'  # redefinition of unused name from line N
+    'F812,'  # list comprehension redefines 'foo' from line N
+    'F821,'  # undefined name 'Foo'
+    'F822,'  # undefined name name in __all__
+    'F823,'  # local variable name … referenced before assignment
+    'F831,'  # duplicate argument name in function definition
+    'F841,'  # local variable 'foo' is assigned to but never used
+    'W191,'  # indentation contains tabs
+    'W291,'  # trailing whitespace
+    'W292,'  # no newline at end of file
+    'W293,'  # blank line contains whitespace
+    'W601,'  # .has_key() is deprecated, use "in"
+    'W602,'  # deprecated form of raising exception
+    'W603,'  # "<>" is deprecated, use "!="
+    'W604,'  # backticks are deprecated, use "repr()"
+    'W605,'  # invalid escape sequence "x"
+    'W606,'  # 'async' and 'await' are reserved keywords starting with Python 3.7
+)
+
+
+def check_dependencies():
+    working_set = {pkg.key for pkg in pkg_resources.working_set}
+
+    for dep in DEPS:
+        if dep not in working_set:
+            print(f"Skipping Python linting since {dep} is not installed.")
+            exit(0)
+
+
+def main():
+    check_dependencies()
+
+    if len(sys.argv) > 1:
+        flake8_files = sys.argv[1:]
+    else:
+        files_args = ['git', 'ls-files', '*.py']
+        flake8_files = subprocess.check_output(files_args).decode("utf-8").splitlines()
+
+    flake8_args = ['flake8', '--ignore=B,C,E,F,I,N,W', f'--select={ENABLED}'] + flake8_files
+    flake8_env = os.environ.copy()
+    flake8_env["PYTHONWARNINGS"] = "ignore"
+
+    try:
+        flake8_out = subprocess.run(flake8_args, stdout=subprocess.PIPE, env=flake8_env)
+    except subprocess.CalledProcessError as e:
+        print(e.output.decode("utf-8"), end="")
+        exit(1)
+
+    print(flake8_out.stdout.decode("utf-8"), end="")
```

Comment:
how is this different from

```suggestion
    try:
        subprocess.check_call(flake8_args, env=flake8_env)
    except subprocess.CalledProcessError:
        exit(1)
```


https://github.com/bitcoin/bitcoin/pull/24794#discussion_r844851739

Reply:
Looking at the docs I got the impression that only `run` allowed to pass in a `env` argument but it seems it works and is just not explicitly named for `check_call`. So I changed this.